test results undates in new branch

IuliiaMaksimova opened the pull request:
1. Обновление структуры проекта: удалила vs.code, вытащила из .gitignore файлы .prettier и .nvmrc
2. Изменила логирование на библиотеку simple logger, обновила все файлы, исходя из этого
3. _Ты так делаешь в тесте Tests/nodejsHomePageTest.js.  Но я не понимаю зачем ты делаешь через this, то есть ты можешь просто сделать переменную logger и её использовать._
_Вот это всё this.currentTest && this.currentTest.isFailed() можно заменить на this.currentTest.isFailed()_
Поправила эти моменты
4. Добавила обработку падений во второй тест

Commit:
test results undates in new branch

Commit:
updates

Commit:
1. Utils: constants, testSteps, afterEach, hook.
2. Updated logger file.
3. Added basePage file with basics methods.
4. Added function of WaitIsPresented.
5. Step by step tests

Nebyt commented on Pages/baseNodePage.js:
```diff
@@ -1,109 +1,78 @@
 
-  setLogger(logger) {
-    this.logger = typeof logger === "function" ? logger : () => {};
+class FooterLocators {
```
Я бы предложил переместить просто переместить все эти локаторы в BaseNodePage, т.к. у этого класса нет никаких специфичных методов и он ничего не наследует.

Nebyt commented on Pages/baseNodePage.js:
```diff
@@ -1,109 +1,78 @@
     return element.isDisplayed();
   }
 
   async clickElement(locator) {
```
Этот метод должен быть в BasePage классе

Nebyt commented on Pages/baseNodePage.js:
```diff
@@ -1,109 +1,78 @@
     await this.driver.wait(until.elementIsVisible(header), 10000);
   }
 
   async isElementVisible(locator) {
```
Этот метод должен быть в BasePage классе

Nebyt commented on Pages/basePage.js:
```diff
@@ -0,0 +1,64 @@
+    this.logger = logger && typeof logger.info === 'function' ? logger : { info: () => {} };
+  }
+
+  async open() {
+    this.logger.info('Base: open https://nodejs.org');
+    await this.driver.get('https://nodejs.org');
+  }
```
У тебя метод `open` реализован в двух местах. Здесь и в классе `NodejsHomePage`, при этом `NodejsHomePage` наследует `BaseNodePage`.
При этом они у тебя практически идентичные с минимальным отличием в линке. То есть ты переписываешь родительский метод.

Я бы предложил изменить текущий метод следующим образом.
Назвать его `openPage` и он должен принимать линку.
```suggestion
  async openPage(url) {
    this.logger.info(`Base: open ${url}`);
    await this.driver.get(url);
  }
```
А в классе `NodejsHomePage` в методе `open` использовать в этот родительский метод `openPage` и просто передавать линку домашней страницы.
Т.е. вот так
```javascript
  async open() {
    await this.openPage('https://nodejs.org/en');
  }
```

Саму же линку на домашнюю страницу, по-хорошему, надо в константы закинуть.

Nebyt commented on Pages/nodejsHomePage.js:
```diff
@@ -1,83 +1,94 @@
     }
   }
 
   async goBack() {
```
Метод `goBack` реализован на всех страницах. Оставь его только на `basePage` на остальных он будет доступен, т.к. они наследуют от `basePage`.

Nebyt commented on Tests/homePage.test.js:
```diff
@@ -0,0 +1,187 @@
+
+    nodeHome = new nodejsHomePage(testSteps.driver, testSteps.logger);
+    downloadPage = new nodejsDownloadPage(testSteps.driver, testSteps.logger);
+    supportPage = new nodejsSupportPage(testSteps.driver, testSteps.logger);
```
Это нигде не используется. Удали если не нужно.

Commit:
Fix test errors and restore beforeEach block in homePage.test.js

- Restored beforeEach block in homePage.test.js to create page objects
- Fixed logger initialization issues
- Updated all page objects to use proper logger.info() calls
- Improved test stability and error handling

Commit:
Comments updates:
1. moved common methods in basePage (open, click, isElementVisible, goBack)
2. update logger function in basePage
3. added new constants: 2,3,5 sec, update files

Commit:
Update test files with new logic of dependency withhook.js file

Nebyt commented on Pages/nodejsHomePage.js:
```diff
@@ -1,83 +1,87 @@
     this.getSupportButton = By.css('a[href$="/en/eol"]');
   }
 
   async open() {
```
Этот метод реализован в родительском классе. Здесь он не нужен.

Nebyt commented on package.json:
```diff
@@ -19,12 +19,13 @@
   "dependencies": {
-    "selenium-webdriver": "^4.35.0"
+    "selenium-webdriver": "^4.35.0",
+    "simple-logger": "^1.0.1",
+    "simple-node-logger": "^21.8.12"
```
Если ты решила не использовать библиотеки, то тебе не нужны они здесь.
